src/sfizz/Voice.cpp

```diff
@@ -581,25 +688,112 @@ void sfz::Voice::fillWithData(AudioSpan<float> buffer) noexcept
+                        xfCurve[i] = clamp(1.0f - xfCoeff[i], 0.0f, 1.0f);
+                }
+                // apply out curve
+                if (0)
+                    ptBuffer.applyGain(xfCurve);
+                else {
+                    // scalar fallback: buffer and curve not aligned
+                    size_t numChannels = ptBuffer.getNumChannels();
+                    for (size_t c = 0; c < numChannels; ++c) {
+                        absl::Span<float> channel = ptBuffer.getSpan(c);
+                        for (unsigned i = 0; i < ptSize; ++i)
+                            channel[i] *= xfCurve[i];
+                    }
+                }
+            }
+            //----------------------------------------------------------------//
+            // Crossfade In
+            //   -> fade in signal preceding the loop start
+            {
+                // compute indices of the crossfade input segment
+                absl::Span<int> xfInIndices = xfadeIndexTemp[0]->first(ptSize);
+                absl::c_copy(ptIndices, xfInIndices.begin());
+                subtract1(loopXfOutStart - loopXfInStart, xfInIndices);
+
+                // disregard the segment whose indices have been pushed
+                // into the negatives, take these virtually as zeroes.
+                unsigned applyOffset = 0;
+                while (applyOffset < ptSize && xfInIndices[applyOffset] < 0)
+                    ++applyOffset;
+                unsigned applySize = ptSize - applyOffset;
+
+                // offset the indices
+                xfInIndices = xfInIndices.subspan(applyOffset);
+                // offset the coeffs
+                absl::Span<float> xfInCoeff = xfCoeff.subspan(applyOffset);
+                // offset the output buffer
+                AudioSpan<float> xfInBuffer = ptBuffer.subspan(applyOffset);
+
+                // compute in curve
+                absl::Span<float> xfCurve = xfadeTemp[1]->first(applySize);
+                IF_CONSTEXPR (loopXfadeUseCurves == 2) {
+                    const Curve& xfIn = getSCurve();
+                    for (unsigned i = 0; i < applySize; ++i)
+                        xfCurve[i] = xfIn.evalNormalized(xfInCoeff[i]);
+                }
+                else IF_CONSTEXPR (loopXfadeUseCurves == 1) {
+                    const Curve& xfIn = resources.curves.getCurve(5);
+                    for (unsigned i = 0; i < applySize; ++i)
+                        xfCurve[i] = xfIn.evalNormalized(xfInCoeff[i]);
+                }
+                else IF_CONSTEXPR (loopXfadeUseCurves == 0) {
+                    // TODO(jpc) vectorize this
+                    for (unsigned i = 0; i < applySize; ++i)
+                        xfCurve[i] = clamp(xfInCoeff[i], 0.0f, 1.0f);
+                }
+                // apply in curve
+                fillInterpolatedWithQuality<true>(
+                    source, xfInBuffer, xfInIndices, *coeffs, xfCurve, quality);
```

Comment:
Should it be `xfInCoeff` here instead of `*coeffs`?

Reply:
No, it should be `ptCoeffs->subspan(applyOffset)`.
these other "coeff" tables are indeed confusing, I have to rename these.
They are the position of the fade on the curve.